cmd/argocd/commands/app.go

```diff
@@ -1620,12 +1683,27 @@ func waitOnApplicationStatus(acdClient apiclient.Client, appName string, timeout
 	_, _ = fmt.Fprintf(w, waitFormatString, "TIMESTAMP", "GROUP", "KIND", "NAMESPACE", "NAME", "STATUS", "HEALTH", "HOOK", "MESSAGE")
 
 	prevStates := make(map[string]*resourceState)
-	appEventCh := acdClient.WatchApplicationWithRetry(ctx, appName)
-	conn, appClient := acdClient.NewApplicationClientOrDie()
+	appEventCh := opts.client.WatchApplicationWithRetry(ctx, opts.appName)
+	conn, appClient := opts.client.NewApplicationClientOrDie()
 	defer util.Close(conn)
-	app, err := appClient.Get(ctx, &applicationpkg.ApplicationQuery{Name: &appName})
+	app, err := appClient.Get(ctx, &applicationpkg.ApplicationQuery{Name: &opts.appName})
 	errors.CheckError(err)
 
+	if opts.watch.delete {
+		go func() {
+			for {
+				time.Sleep(3 * time.Second)
+				_, err := appClient.Get(context.Background(), &applicationpkg.ApplicationQuery{Name: &opts.appName})
+				if err != nil {
+					if strings.HasSuffix(err.Error(), "not found") {
+						delete <- true
+						cancel()
+					}
+				}
+			}
+		}()
+	}
```

Comment:
Starting a separate goroutine is not necessary here. We already have a channel below (`appEventCh`) listening to new application events.